packages/cloudflare/src/cli/utils/run-wrangler.ts

```diff
@@ -12,18 +15,37 @@ type WranglerOptions = {
   logging?: "all" | "error";
 };
 
+/**
+ * Checks the package.json `packageManager` field to determine whether yarn modern is used.
+ *
+ * @param options Build options.
+ * @returns Whether yarn modern is used.
+ */
+function isYarnModern(options: BuildOptions) {
+  const packageJson: { packageManager?: string } = JSON.parse(
+    readFileSync(path.join(options.monorepoRoot, "package.json"), "utf-8")
+  );
+
+  if (!packageJson.packageManager?.startsWith("yarn")) return false;
+
+  const [, version] = packageJson.packageManager.split("@");
+  if (!version) return false;
+
+  return compareSemver(version, ">=", "4.0.0");
```

Comment:
```suggestion
  return version ? compareSemver(version, ">=", "4.0.0") : false;
```

What if version is ^4, ~4, 4.x?

Reply:
> What if version is ^4, ~4, 4.x?

Maybe it's always a fixed version there?